packages/grafana-ui/src/components/SetInterval/SetInterval.tsx

```diff
@@ -23,25 +23,41 @@ export class SetInterval extends PureComponent<Props> {
   }
 
   componentDidMount() {
+    // Creating a subscription to propsSubject. This subject pushes values every time
+    // SetInterval's props change
     this.subscription = this.propsSubject
       .pipe(
+        // switchMap creates a new observables based on the input stream,
+        // which becomes part of the propsSubject stream
```

Comment:
There's one very important thing that SwitchMap does and that is that it calls unsubscribe on the previous Observable so that we don't get interval after interval observables added.